src/target/imxrt.c

```diff
@@ -59,21 +59,64 @@
 #define IMXRT_MPU_BASE UINT32_C(0xe000ed90)
 #define IMXRT_MPU_CTRL (IMXRT_MPU_BASE + 0x04U)
 
-#define IMXRT_CCM_ANALOG_BASE     UINT32_C(0x400d8000)
-#define IMXRT_CCM_ANALOG_PLL3_PFD (IMXRT_CCM_ANALOG_BASE + 0x0f0U)
+#define IMXRT_10XX_CCM_ANALOG_BASE     UINT32_C(0x400d8000)
+#define IMXRT_10XX_CCM_ANALOG_PLL3_PFD (IMXRT_10XX_CCM_ANALOG_BASE + 0x0f0U)
 
-#define IMXRT_CCM_ANALOG_PLL_PFD0_FRAC_MASK 0xffffffc0
+#define IMXRT_10XX_CCM_ANALOG_PLL_PFD0_FRAC_MASK 0xffffffc0
 
-#define IMXRT_CCM_BASE  UINT32_C(0x400fc000)
-#define IMXRT_CCM_CSCM1 (IMXRT_CCM_BASE + 0x01cU)
-#define IMXRT_CCM_CCG6  (IMXRT_CCM_BASE + 0x080U)
+#define IMXRT_10XX_CCM_BASE  UINT32_C(0x400fc000)
+#define IMXRT_10XX_CCM_CSCM1 (IMXRT_10XX_CCM_BASE + 0x01cU)
+#define IMXRT_10XX_CCM_CCG6  (IMXRT_10XX_CCM_BASE + 0x080U)
 
-#define IMXRT_CCM_CSCM1_FLEXSPI_CLK_SEL_MASK      0xfc7fffffU
-#define IMXRT_CCM_CSCM1_FLEXSPI_CLK_SEL_PLL3_PFD0 0x03800000U
-#define IMXRT_CCM_CCG6_FLEXSPI_CLK_MASK           0xfffff3ffU
-#define IMXRT_CCM_CCG6_FLEXSPI_CLK_ENABLE         0x00000c00U
+#define IMXRT_10XX_CCM_CSCM1_FLEXSPI_CLK_SEL_MASK      0xfc7fffffU
+#define IMXRT_10XX_CCM_CSCM1_FLEXSPI_CLK_SEL_PLL3_PFD0 0x03800000U
+#define IMXRT_10XX_CCM_CCG6_FLEXSPI_CLK_MASK           0xfffff3ffU
+#define IMXRT_10XX_CCM_CCG6_FLEXSPI_CLK_ENABLE         0x00000c00U
+
+#define IMXRT_11XX_CCM_BASE UINT32_C(0x40CC0000)
+#define IMXRT_11XX_CCM_CLOCK_ROOT20_CONTROL (IMXRT_11XX_CCM_BASE + 0x0 + (20 * 0x80))
+#define IMXRT_11XX_CCM_CLOCK_ROOT20_CONTROL_PLL_480M (0x7 << 8)
+#define IMXRT_11XX_CCM_CLOCK_ROOT20_CONTROL_OSC400M (0x2 << 8)
+#define IMXRT_11XX_CCM_CLOCK_ROOT20_CONTROL_DIV(divisor) ((divisor - 1) << 0)
+#define IMXRT_11XX_CCM_LPCG28 (IMXRT_11XX_CCM_BASE + 0x6000 + (28 * 0x20))
+
+#define SCB_CPUID UINT32_C(0xE000ED00)
+#define SCB_CPUID_PARTNO_Pos                4U                                            /*!< SCB CPUID: PARTNO Position */
+#define SCB_CPUID_PARTNO_Msk               (0xFFFUL << SCB_CPUID_PARTNO_Pos)              /*!< SCB CPUID: PARTNO Mask */
+
+#define CORE_ID_CM33 (0xD21)
+#define CORE_ID_CM7  (0xC27)
```

Comment:
These core IDs are already defined in the cortexm.h header - https://github.com/blackmagic-debug/blackmagic/blob/718c3aedd49224f03ee4f34cb9b2668b0311fa03/src/target/cortexm.h#L178-L185

Please make use of them rather than re-defining them here.